google_guest_agent/network/manager/manager.go

```diff
@@ -0,0 +1,78 @@
+//  Copyright 2024 Google LLC
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+// Package manager is responsible for detecting the current network manager service, and
+// writing and rolling back appropriate configurations for each network manager service.
+package manager
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/cfg"
+	"github.com/GoogleCloudPlatform/guest-agent/metadata"
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+)
+
+// Service is an interface for setting up network configurations
+// using different network managing services, such as systemd-networkd and wicked.
+type Service interface {
+	// IsManaging checks whether this network manager service is managing the provided interface.
+	IsManaging(ctx context.Context, iface string) (bool, error)
+
+	// Name is the name of the network manager service.
+	Name() string
+
+	// Setup writes the appropriate configurations for the network manager service for all
+	// non-primary network interfaces.
+	Setup(ctx context.Context, config *cfg.Sections, payload []metadata.NetworkInterfaces) error
+
+	// Rollback rolls back the changes created in Setup.
+	Rollback(ctx context.Context, payload []metadata.NetworkInterfaces) error
+}
+
+var (
+	googleComment = "# Added by Google Compute Engine Guest Agent."
```

Comment:
nit: these should be constants instead of variables (except `knownNetworkManagers`)